core/src/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -1113,7 +1113,9 @@ public String getFormatMask() {
 
     if ( !Utils.isEmpty( this.conversionMask ) ) {
       mask = this.conversionMask;
-    }
+    } else {
```

Comment:
This function is at the *base class* so there is no guarantee that the default mask to apply should be the DEFAULT_DATE_MASK.

To this specific case we could override this method in the String meta value class to provide the default date mask if it was doing the String -> Date conversion